docs/nav2_overview.md

```diff
@@ -0,0 +1,146 @@
+
+[BehaviorTree.CPP][behavior_tree_cpp] is a C++ library for working with behavior trees. The library is essentially a
+lightweight task orchestrator; it decides which actions to take, but the actual computation gets dispatched to
+separate threads.
+
+While the BehaviorTree.CPP library is independent from ROS, Nav2 extends it with custom BT Action nodes (and non-Action
+nodes). The custom BT Action nodes provide a ROS-compatible interface, allowing the library to dispatch work via ROS 2
+actions.
+
+[behavior_tree_wiki]: https://en.wikipedia.org/wiki/Behavior_tree_(artificial_intelligence,_robotics_and_control)
+[behavior_tree_cpp]: https://www.behaviortree.dev/
+
+## Behavior Tree (BT) Navigator Server
+
+This server contains _navigators_: decision makers that manage calls to the other servers to provide high-level
+autonomy. Nav2 comes with two navigators: `NavigateToPose` and `NavigateThroughPoses`. Both navigators load behavior
+trees from XML files, and users can customize each navigator's functionality through these XML files. Additionally,
+users can provide their own navigators if they desire.
+
+For an example, consider the following behavior tree specification:
+
+```xml
+<root BTCPP_format="4" main_tree_to_execute="MainTree">
+  <BehaviorTree ID="MainTree">
+    <PipelineSequence name="NavigateWithReplanning">
+      <ControllerSelector selected_controller="{selected_controller}" default_controller="FollowPath" topic_name="controller_selector"/>
+      <PlannerSelector selected_planner="{selected_planner}" default_planner="GridBased" topic_name="planner_selector"/>
+      <DistanceController distance="1.0">
+        <ComputePathToPose goal="{goal}" path="{path}" planner_id="{selected_planner}" error_code_id="{compute_path_error_code}"/>
+      </DistanceController>
+      <FollowPath path="{path}" controller_id="{selected_controller}" error_code_id="{follow_path_error_code}"/>
+    </PipelineSequence>
+  </BehaviorTree>
+</root>
+```
+
+From this specification, the `NavigateToPose` navigator will behave as follows:
+
+1. Make a service call to the Controller Server to activate the specified controller plugin. (See `params.yaml` for
+   more information.)
+2. Make a service call to the Planner Server to activate to specified planner plugin. (See `params.yaml` for more
+   information.)
+3. Make an action call to the Planner Server to generate a new reference path to track. The navigator will request a
+   new reference path whenever the robot is greater than 1 meter from the goal pose.
+4. Make an action call to the Controller Server to track the reference path.
+
+The navigator will repeat the pipeline until the robot reaches its gaol pose.
+
+> [!NOTE]
+> The Planning and Control Servers' plugins do the actual planning and tracking, respectively. The navigator, which
+> lives within the BT Navigator Server, only decides when to send the corresponding ROS 2 action requests.
+
+## Planning Server
```

Comment:
I noticed in this document sometimes we refer to it as `Planning Server` and sometimes as `Planner Server'. I think its clear, but probably best to remain consistent if there is an official name Nav2 typically uses.